test/internal/environment/environment.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package environment
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/tektoncd/pipeline/test/internal/clients"
+	"k8s.io/apimachinery/pkg/version"
+	knativetest "knative.dev/pkg/test"
+)
+
+var skipRootUserTests = false
+
+func init() {
+	flag.BoolVar(&skipRootUserTests, "skipRootUserTests", false, "Skip tests that require root user")
+}
+
+// Execution contains information about the current test execution and daemon
+// under test
+type Execution struct {
+	ConfigPath  string
+	ClusterName string
+	Clients     *clients.Clients
+	Info        *version.Info
+	Platform    string
+
+	koDockerRepo      string
+	skipRootUserTests bool
+}
+
+// New creates a new Execution struct
+// This is configured using the env client (see client.FromEnv)
+func New() (*Execution, error) {
+	c, err := clients.NewClients(knativetest.Flags.Kubeconfig, knativetest.Flags.Cluster, "default")
+	if err != nil {
+		return nil, err
+	}
+	v, err := c.KubeClient.Kube.Discovery().ServerVersion()
+	if err != nil {
+		return nil, err
+	}
+	return &Execution{
+		Clients:     c,
+		Info:        v,
+		Platform:    v.Platform,
```

Comment:
I really like this option to get `platform` from actual target k8s cluster, so no need any more to use `TEST_RUNTIME_ARCH`, in case when tests are initiated on machine of one arch, but will be executed on k8s cluster with another arch 👍 
Can you also remove mention about this env variable from [doc](https://github.com/tektoncd/pipeline/tree/master/test#running) ?

Reply:
we can yes :upside_down_face: